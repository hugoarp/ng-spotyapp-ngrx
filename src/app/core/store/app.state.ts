import { ActionReducerMap } from '@ngrx/store';
import { TrackInfoState } from '../interfaces/track';
import { UserInfoState } from '../interfaces/user';
import { favoritesTrackInfoReducer } from './favorites/favorites.reducers';
import { recommendedTrackInfoReducer } from './recommended/recommended.reducers';
import { userInfoReducer } from './user/user.reducers';

export interface AppState {
  userInfo: UserInfoState;
  recommendedTrackInfo: TrackInfoState;
  favoritesTrackInfo: TrackInfoState;
}

export const ROOT_REDUCERS: ActionReducerMap<AppState> = {
  userInfo: userInfoReducer,
  recommendedTrackInfo: recommendedTrackInfoReducer,
  favoritesTrackInfo: favoritesTrackInfoReducer,
};
