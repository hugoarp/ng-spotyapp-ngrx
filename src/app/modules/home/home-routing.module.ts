import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { HomeComponent } from './page/home.component';

const routes: Routes = [
  {
    path: '',
    component: HomeComponent,
    children: [
      {
        path: 'favoritos',
        loadChildren: () =>
          import('../favorites/favorites.module').then(
            (m) => m.FavoritesModule
          ),
      },
      {
        path: 'recomendados',
        loadChildren: () =>
          import('../recommended/recommended.module').then(
            (m) => m.RecommendedModule
          ),
      },
      { path: '', redirectTo: 'recomendados', pathMatch: 'full' },
    ],
  },
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class HomeRoutingModule {}
