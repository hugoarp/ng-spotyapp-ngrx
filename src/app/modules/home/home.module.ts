import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { HomeRoutingModule } from './home-routing.module';
import { HomeComponent } from './page/home.component';
import { RouterModule } from '@angular/router';
import { OrganismsModule } from 'src/app/shared/components/organisms/organisms.module';
import { MoleculesModule } from 'src/app/shared/components/molecules/molecules.module';

@NgModule({
  declarations: [HomeComponent],
  imports: [
    CommonModule,
    HomeRoutingModule,
    OrganismsModule,
    MoleculesModule,
    RouterModule,
  ],
})
export class HomeModule {}
