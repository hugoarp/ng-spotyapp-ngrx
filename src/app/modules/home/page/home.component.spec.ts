import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { provideMockStore } from '@ngrx/store/testing';
import { HomeComponent } from './home.component';

describe('HomeComponent', () => {
  let component: HomeComponent;
  let fixture: ComponentFixture<HomeComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [HomeComponent],
      providers: [provideMockStore({})],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(HomeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });
});
