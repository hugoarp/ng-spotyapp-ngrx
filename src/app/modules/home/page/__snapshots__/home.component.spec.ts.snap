// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <app-nav-menu />
  <app-user-button />
  <router-outlet />
</div>
`;
